File: payroll-api/src/main/java/com/company/payroll/mapper/StaffResignationMapper.java
package com.company.payroll.mapper;

import java.util.List;

import com.company.payroll.model.StaffResignation;
import org.springframework.stereotype.Repository;

@Repository
public interface StaffResignationMapper {
    int deleteByPrimaryKey(Integer resignId);
    int insert(StaffResignation row);
    int insertSelective(StaffResignation row);
    List<StaffResignation> selectList();
    StaffResignation selectByPrimaryKey(Integer resignId);
    int updateByPrimaryKeySelective(StaffResignation row);
    int updateByPrimaryKey(StaffResignation row);
}

File: kotlin-payroll/src/main/kotlin/com/company/payroll/mapper/StaffResignationMapper.kt
package com.company.payroll.mapper

import com.company.payroll.model.StaffResignation
import org.springframework.stereotype.Repository

@Repository
interface StaffResignationMapper {
  fun deleteByPrimaryKey(resignId: Int): Int

  fun insert(row: StaffResignation): Int

  fun insertSelective(row: StaffResignation): Int

  fun selectByPrimaryKey(resignId: Int): StaffResignation

  fun selectList(): List<StaffResignation>

  fun updateByPrimaryKeySelective(row: StaffResignation): Int

  fun updateByPrimaryKey(row: StaffResignation): Int
}